source/lac/petsc_matrix_base.cc

```diff
@@ -80,6 +80,16 @@ namespace PETScWrappers
   {}
 
 
+  MatrixBase::MatrixBase(const Mat &A)
+    : matrix(A)
+    , last_action(VectorOperation::unknown)
+  {
+    const PetscErrorCode ierr =
+      PetscObjectReference(reinterpret_cast<PetscObject>(matrix));
+    AssertNothrow(ierr == 0, ExcPETScError(ierr));
+    (void)ierr;
```

Comment:
```suggestion
    AssertThrow(ierr == 0, ExcPETScError(ierr));
```
It's OK to throw an exception from a ctor